cms/djangoapps/contentstore/views/component.py

```diff
@@ -189,57 +257,92 @@ def create_template_dict(name, cat, boilerplate_name=None, tab="common", hinted=
     if library:
         component_types = [component for component in component_types if component != 'discussion']
 
+    component_types = _filter_disabled_blocks(component_types)
+
+    # Content Libraries currently don't allow opting in to unsupported xblocks/problem types.
+    allow_unsupported = getattr(courselike, "allow_unsupported_xblocks", False)
+
     for category in component_types:
+        authorable_variations = authorable_xblocks(allow_unsupported=allow_unsupported, name=category)
+        support_level_without_template = component_support_level(authorable_variations, category)
         templates_for_category = []
         component_class = _load_mixed_class(category)
-        # add the default template with localized display name
-        # TODO: Once mixins are defined per-application, rather than per-runtime,
-        # this should use a cms mixed-in class. (cpennington)
-        display_name = xblock_type_display_name(category, _('Blank'))  # this is the Blank Advanced problem
-        templates_for_category.append(create_template_dict(display_name, category, None, 'advanced'))
-        categories.add(category)
+
+        if support_level_without_template:
```

Comment:
Note to reviewers: to easily see what changed here, add "?w=1" to the end of the github URL (ignore whitespace changes).